code/modules/hydroponics/biogenerator.dm

```diff
@@ -19,6 +19,42 @@
 /obj/machinery/biogenerator/Initialize(mapload)
 	. = ..()
 	stored_research = new /datum/techweb/specialized/autounlocking/biogenerator
+	register_context()
+
+/obj/machinery/biogenerator/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+
+	if(!isliving(user))
+		return .
+
+	if(!Adjacent(user))
+		return .
+
+	switch (held_item?.tool_behaviour)
+		if (TOOL_SCREWDRIVER)
+			context[SCREENTIP_CONTEXT_LMB] = panel_open ? "Close panel" : "Open panel"
+			return CONTEXTUAL_SCREENTIP_SET
+		if (TOOL_CROWBAR)
+			if (panel_open)
+				context[SCREENTIP_CONTEXT_LMB] = "Remove electronics"
+				return CONTEXTUAL_SCREENTIP_SET
+
+	if (istype(held_item, /obj/item/food/grown/ ))
+		context[SCREENTIP_CONTEXT_LMB] = "Insert product"
+		return CONTEXTUAL_SCREENTIP_SET
+	else if (istype(held_item, /obj/item/storage/bag/plants))
+		context[SCREENTIP_CONTEXT_LMB] = "Insert products"
+		return CONTEXTUAL_SCREENTIP_SET
+	else if (istype(held_item, /obj/item/reagent_containers/glass ))
+		context[SCREENTIP_CONTEXT_LMB] =  beaker ? "Swap container" : "Insert container"
+		return CONTEXTUAL_SCREENTIP_SET
+	else if (istype(held_item, /obj/item/ ))
```

Comment:
is this needed at all?

Reply:
Not necessarily. It's more to accurately hint what's exactly happening if you try to insert a container when the biogenerator  already has a beaker. 
And for the last condition, that's for handling the rest of items that don't directly have an interaction with the machine. If you are holding a chair and you try to left click the machine, nothing happens. So it makes sense the screentip doesn't provide an indication.
After we have checked all the items in hands, if we haven't done a return yet, we could assume we are then bare handed, in which the code continues.